File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/group_capsule/data/dto/GroupCapsuleCreateRequestDto.java
package site.timecapsulearchive.core.domain.capsule.group_capsule.data.dto;

import java.time.ZonedDateTime;
import java.util.List;
import lombok.Builder;
import org.locationtech.jts.geom.Point;
import site.timecapsulearchive.core.domain.capsule.entity.Capsule;
import site.timecapsulearchive.core.domain.capsule.entity.CapsuleType;
import site.timecapsulearchive.core.domain.capsuleskin.entity.CapsuleSkin;
import site.timecapsulearchive.core.domain.group.entity.Group;
import site.timecapsulearchive.core.domain.member.entity.Member;
import site.timecapsulearchive.core.infra.map.data.dto.AddressData;

@Builder
public record GroupCapsuleCreateRequestDto(
    List<Long> groupMemberIds,
    List<String> imageNames,
    List<String> videoNames,
    Long capsuleSkinId,
    String title,
    String content,
    Double longitude,
    Double latitude,
    AddressData addressData,
    ZonedDateTime dueDate
) {

    public Capsule toGroupCapsule(
        Member member,
        CapsuleSkin capsuleSkin,
        Group group,
        CapsuleType capsuleType,
        Point point
    ) {
        return Capsule.builder()
            .title(title)
            .content(content)
            .point(point)
            .address(addressData.toAddress())
            .type(capsuleType)
            .member(member)
            .dueDate(dueDate)
            .capsuleSkin(capsuleSkin)
            .group(group)
            .build();
    }
}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/group_capsule/GroupCapsuleCreateRequestDto.kt
package com.droidblossom.archive.data.dto.group_capsule

import com.droidblossom.archive.data.dto.capsule.response.AddressDataDto

data class GroupCapsuleCreateRequestDto(
    val groupMemberIds : List<Long>,
    val capsuleSkinId: Long,
    val content: String,
    val directory: String,
    val dueDate: String?,
    val imageNames: List<String>?,
    val videoNames: List<String>?,
    val addressData: AddressDataDto,
    val latitude: Double,
    val longitude: Double,
    val title: String
)